.github/workflows/sanity-test.yaml

```diff
@@ -0,0 +1,314 @@
+name: Sanity Testing
+on: 
+  push:
+  # TODO: uncomment before merging
+    # branches:
+    #   - main
+  workflow_dispatch:
+
+
+
+permissions:
+  # required to retrieve AWS credentials
+  id-token: write
+  contents: write
+
+# cancel currently running jobs if a new version of the branch is pushed
+concurrency:
+  group: sanity_testing-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+
+jobs:
+  Sanity-Tests:
+    environment: Testing
+    runs-on: ubuntu-latest
+    env:
+      AZURE_CLIENT_ID: ${{ secrets.CLIENT_ID }}
+      AZURE_CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
+      AZURE_TENANT_ID: ${{ secrets.TENANT_ID }}
+      CORSO_M365_TEST_USER_ID: ${{ secrets.CORSO_M365_TEST_USER_ID }}
+      CORSO_PASSPHRASE: ${{ secrets.INTEGRATION_TEST_CORSO_PASSPHRASE }}
+      TEST_RESULT: "test_results"
+      CORSO_BUCKET: ${{ secrets.CI_TESTS_S3_BUCKET }}
+
+    defaults:
+      run:
+        working-directory: src
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Golang with cache
+        uses: magnetikonline/action-golang-cache@v3
+        with:
+          go-version-file: src/go.mod
+
+      - run: make build
+
+      - run: go build -o sanityCheck ./cmd/sanity_test
+
+      - run: mkdir test_results
+
+      # AWS creds
```

Comment:
I missed it earlier, but this can be a one time thing. As of now, we seem to be doing this after every test.

Reply:
Done